File: domain-driven-design/java-web/src/main/java/com/javaweborder/domain/order/OrderRepository.java
// 领域层(Domain)：订单数据仓库接口
package com.javaweborder.domain.order;

import com.javaweborder.infrastructure.repository.Repository;

import java.util.List;
import java.util.Optional;

// OrderRepository 订单仓储接口，继承通用 Repository<Order>
public interface OrderRepository extends Repository<Order> {
    
    // 保存订单
    boolean save(Order order);

    // 根据ID查找订单
    Optional<Order> findById(int id);

    // 查找所有订单
    List<Order> findAll();

    // 删除订单
    void delete(int id);

    // 根据客户名称查找订单
    List<Order> findByCustomerName(String customerName);
}

File: domain-driven-design/go-web/internal/domain/repository/order_repository.go
// 领域层(Domain)：订单数据仓库接口
package repository

import (
	"go-web-order/internal/domain/order"
)

// OrderRepository 订单仓储接口，定义对订单数据的操作。
// type OrderRepository interface {
//   Save(order *order.Order) error                                  // 保存订单
//   FindByID(id int64) (*order.Order, error)                        // 根据ID查找订单
//   FindAll(userId int) ([]*order.Order, error)                     // 查找所有订单
//   Delete(id int64) error                                          // 删除订单
//   FindByCustomerName(customerName string) ([]*order.Order, error) // 根据客户名称查找订单
// }

// [可选]OrderRepository 订单仓储接口，继承通用仓储接口 Repository[order.Order]
type OrderRepository interface {
  Repository[order.Order]  // 继承通用仓储接口
  FindByCustomerName(customerName string) ([]*order.Order, error) // 根据客户名称查找订单
}
